### PR TITLE
Add h7 power management

### PR DESCRIPTION
**Pull request by benediktibk**

### CODE REVIEW

**Comment by benediktibk** [comment]
Requires https://github.com/zephyrproject-rtos/zephyr/pull/41869 to be completed beforehand.

**Comment by benediktibk** [comment]
Also requires https://github.com/zephyrproject-rtos/zephyr/pull/41867 to be completed beforehand.

File: soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series [hunk 1]
```diff
@@ -16,6 +17,9 @@ config ROM_START_OFFSET
 	default 0x400 if BOOTLOADER_MCUBOOT
 	default 0x0   if !BOOTLOADER_MCUBOOT
 
+config STM32_LPTIM_TIMER
+	default y if PM
+
```
Comment (erwango) [inline soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series:22]: Move this to https://github.com/zephyrproject-rtos/zephyr/pull/41867

File: boards/arm/stm32h735g_disco/stm32h735g_disco.dts [hunk 2]
```diff
@@ -113,3 +137,11 @@
 	pinctrl-names = "default";
 	cd-gpios = <&gpiof 5 GPIO_ACTIVE_LOW>;
 };
+
+&lptim1 {
+	status = "okay";
+};
```
Comment (erwango) [inline boards/arm/stm32h735g_disco/stm32h735g_disco.dts:143]: Please move this to https://github.com/zephyrproject-rtos/zephyr/pull/41867

**Comment by benediktibk** [comment]
In the project where we were using the H7 the requirement for power saving was dropped. On top of that, additional tasks for getting the power saving up and running popped up, like having to change clock frequency dependening on the VOS-level, ... and so on. Unfortunately getting power saving right is rather difficult.

**Comment by semihalf-niedzwiecki-dawid** [comment]
Ok, thanks for the info. Does it need some improvements to be merged? At the first glance, it looks like ready to merge.

**Comment by benediktibk** [comment]
I haven't finally tested it in this state, with some parts being stripped out into other PRs. Therefore, I would heavily advise for testing it on the actual hardware first.

**Pull request closed without merging**